examples/jsm/loaders/SVGLoader.js

```diff
@@ -1680,6 +1684,463 @@ SVGLoader.pointsToStroke = function ( points, style, arcDivisions, minDistance )
+		DESTINATION: 1,
+		BETWEEN: 2,
+		LEFT: 3,
+		RIGHT: 4,
+		BEHIND: 5,
+		BEYOND: 6
+	};
+
+	function findEdgeIntersection( edge1, edge2 ) {
+
+		var x1 = edge1[ 0 ].x;
+		var x2 = edge1[ 1 ].x;
+		var x3 = edge2[ 0 ].x;
+		var x4 = edge2[ 1 ].x;
+		var y1 = edge1[ 0 ].y;
+		var y2 = edge1[ 1 ].y;
+		var y3 = edge2[ 0 ].y;
+		var y4 = edge2[ 1 ].y;
+		var nom1 = ( x4 - x3 ) * ( y1 - y3 ) - ( y4 - y3 ) * ( x1 - x3 );
+		var nom2 = ( x2 - x1 ) * ( y1 - y3 ) - ( y2 - y1 ) * ( x1 - x3 );
+		var denom = ( y4 - y3 ) * ( x2 - x1 ) - ( x4 - x3 ) * ( y2 - y1 );
+		var t1 = nom1 / denom;
+		var t2 = nom2 / denom;
+		var interPoints = [];
+
+		if ( ( ( denom === 0 ) && ( nom1 !== 0 ) ) || ( t1 <= 0 ) || ( t1 >= 1 ) || ( t2 < 0 ) || ( t2 > 1 ) ) {
+
+			//1. lines are parallel or edges don't intersect
+
+			return interPoints;
+
+		} else if ( ( nom1 === 0 ) && ( denom === 0 ) ) {
+
+			//2. lines are colinear
+
+			//check if endpoints of edge2 lies on edge1
+			for ( var i = 0; i < 2; i ++ ) {
+
+				var classify = classifyPoint( edge2[ i ], edge1 );
+				//find position of this endpoints relatively to edge1
+				if ( classify.loc == IntersectionLocationType.ORIGIN || classify.loc == IntersectionLocationType.DESTINATION ) {
+
+					interPoints.push( { x: edge2[ i ].x, y: edge2[ i ].y, t: classify.t } );
+
+				} else if ( classify.loc == IntersectionLocationType.BETWEEN ) {
+
+					x = + ( ( x1 + classify.t * ( x2 - x1 ) ).toPrecision( 10 ) );
+					y = + ( ( y1 + classify.t * ( y2 - y1 ) ).toPrecision( 10 ) );
+					interPoints.push( { x: x, y: y, t: classify.t } );
+
+				}
+
+			}
+
+			return interPoints;
+
+		} else {
+
+			//3. edges intersect
+
+			for ( var i = 0; i < 2; i ++ ) {
+
+				var classify = classifyPoint( edge2[ i ], edge1 );
+
+				if ( classify.loc == IntersectionLocationType.ORIGIN || classify.loc == IntersectionLocationType.DESTINATION ) {
+
+					interPoints.push( { x: edge2[ i ].x, y: edge2[ i ].y, t: classify.t } );
+
+				}
+
+			}
+
+			if ( interPoints.length > 0 ) {
+
+				return interPoints;
+
+			}
+
+			var x = + ( ( x1 + t1 * ( x2 - x1 ) ).toPrecision( 10 ) );
+			var y = + ( ( y1 + t1 * ( y2 - y1 ) ).toPrecision( 10 ) );
+			interPoints.push( { x: x, y: y, t: t1 } );
+
+			return interPoints;
+
+		}
+
+	}
+
+	function classifyPoint( p, edge ) {
+
+		var ax = edge[ 1 ].x - edge[ 0 ].x;
+		var ay = edge[ 1 ].y - edge[ 0 ].y;
+		var bx = p.x - edge[ 0 ].x;
+		var by = p.y - edge[ 0 ].y;
+		var sa = ax * by - bx * ay;
+
+		if ( ( p.x === edge[ 0 ].x ) && ( p.y === edge[ 0 ].y ) ) {
+
+			return { loc: IntersectionLocationType.ORIGIN, t: 0 };
+
+		}
+
+		if ( ( p.x === edge[ 1 ].x ) && ( p.y === edge[ 1 ].y ) ) {
+
+			return { loc: IntersectionLocationType.DESTINATION, t: 1 };
+
+		}
+
+		var theta = ( polarAngle( [ edge[ 1 ], edge[ 0 ] ] ) -
+			polarAngle( [ { x: edge[ 1 ].x, y: edge[ 1 ].y }, { x: p.x, y: p.y } ] ) ) % 360;
+
+		if ( theta < 0 ) {
+
+			theta = theta + 360;
+
+		}
+
+		if ( sa < - 0.0000000001 ) {
```

Comment:
You can probably use `Number.EPSILON` here?

Reply:
I'll try that. Do you also have such a Number for my `BIGNUMBER` constant? I tried `Number.MAX_SAFE_INTEGER` but then the algorithm acts up.